pilot/pkg/xds/endpoint_builder.go

```diff
@@ -106,6 +110,9 @@ func (b EndpointBuilder) Key() string {
 	if b.service != nil {
 		params = append(params, string(b.service.Hostname)+"/"+b.service.Attributes.Namespace)
 	}
+	for _, pa := range b.peerAuthentications {
+		params = append(params, pa.Name+"/"+pa.Namespace)
```

Comment:
do we have concerns about the size of this list? When I was looking at CDS caching we need a similar thing for Envoyfilter and I was concerned. I hypothesized a compact representation (like a bitmap, where we have a bit for each peer authentication and set 1 if it applies to this proxy). Thinking about it now, that seems over complex - worst case we get a long string which probably isn't too bad? WDYT?

Reply:
I'm not too concerned short-term. The next logical follow-up is probably looking at workloadSelector to trim it down.